### DISCUSSION TITLE
WatsonWsClient set IP

### DISCUSSION THREAD

**Discussion by WrongRust** [jchristn/WatsonWebsocket#126] [Q&A]
Good day, I have a question, when creating WatsonWsClient, how can I specify my IP to it, it is required to bind clients.
Now he takes the IP from which he connects

**Comment by jchristn** [comment]
Hi @WrongRust you can't do this in code.  The IP address from which a client connection is established is determined by the operating system based on the available set of network interfaces (and their IP addresses) and the network/host address to which you wish to connect.

**Comment by WrongRust** [comment]
For example, in UDP it is possible to specify the IP from which the request is created (UdpClient = new UdpClient(new IPEndPoint(), 0))). Also in the Socket there is a Bind where the IP is indicated. I thought maybe in your library there is such an opportunity

**Reply by jchristn** [reply]
I wish that were the case.  This library rides on top of http.sys, which handles all of the socket work on our behalf.

**Reply by WrongRust** [reply]
I just need to control the number of requests from a user from different IPs. And my mind has no ideas(
